src/domains/job/common/components/JobInfiniteScroll.jsx

```
@@ -1,70 +1,43 @@
-  const [loading, setLoading] = useState(false)
-  const [hasMore, setHasMore] = useState(true)
   const navigate = useNavigate()
 
-  const loadPosts = useCallback(async () => {
-    if (loading || !hasMore) return
-    setLoading(true)
-    try {
-      const response = await fetcher(lastId, order)
-      let newPosts = response?.[dataKey] || []
-      const newLastId = response?.lastId
+  const { data, fetchNextPage, hasNextPage, isFetchingNextPage, isLoading } = useInfiniteQuery({
```

ESLint

[error] 'isLoading' is assigned a value but never used.

(no-unused-vars)